src/game.cpp

```diff
@@ -3853,25 +3851,39 @@ void Game::updateFrame(std::vector<aabb3f> &highlight_boxes,
 		Inventory
 	*/
 
-	if (client->getPlayerItem() != runData->new_playeritem)
+	bool inventory_selection_changed = false;
+	bool inventory_changed = false;
```

Comment:
I wouldn't be that sure. [section 7.6.8 paragraph 10](http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1256.pdf#page=138&zoom=auto,-27,816) of the C standard handles objects with "automatic storage duration" (like this one), and states that values of such uninitialized objects are indeterminate.